tracks/python/exercises/saddle-points/mentoring.md

```diff
@@ -0,0 +1,118 @@
+# Mentoring
+
+## Problem and challenges
+
+This problem requires the student to find the points in a matrix (represented as `list[list[int]]`) that are both the highest value in that row and the lowest value in that column.
+
+## Reasonable solution
+
+```python
+def saddle_points(matrix):
+    result = []
+    if not matrix:
+        return result
+    if not all(len(row) == len(matrix[0]) for row in matrix):
+        raise ValueError("irregular matrix")
+    for row_num, row in enumerate(matrix):
+        for column_num in range(len(row)):
+            if max(row) == min(row[column_num] for row in matrix):
+                result.append({"row": row_num + 1, "column": column_num + 1})
+    return result
+```
+
+## Common suggestions
+
+### Checking for an empty or invalid matrix
+
+This problem requires identifying when the matrix is irregular i.e. the number of items in each row is inconsistent. 
+The easiest way to do this is to simply check that the length of each row is equal to the length of the first row. 
+Students may also include additional checks for other types of invalid data if desired.
+
+It is also helpful to first identify when an empty matrix has been given and return an empty result. 
+This prevents an `IndexError` from trying to index into an empty list.
+
+```python
+# Best solution2
+if not matrix:
+    return []
+if not all(len(row) == len(matrix[0]) for row in matrix):
+    raise ValueError("irregular matrix")```
+# or:
+if any(len(row) != len(matrix[0]) for row in matrix):
+    raise ValueError("irregular matrix")```
+
+        
+# Acceptable solution
+if len(matrix) == 0:
+  return []
+row_length = len(matrix[0])
+for row in matrix:
+  if len(row) != row_length:
+    raise ValueError("irregular matrix")
+ ```
+
+This might be a good place to introduce concepts like list comprehensions and the `all()` or `any()` functions, which are easier to use here but will be helpful for the main problem.
+
+### Iterating over the values
+
+There are various ways of iterating over the values, but nested `for` loops are the most straightforward. 
+Since the column and row indicies are required along with the values, this is a good place to introduce the `enumerate()` function. 
```

Comment:
```suggestion
Since the column and row indices are required along with the values, this is a good place to introduce the `enumerate()` function. 
```